### DISCUSSION TITLE
REALITY Blocked in Iran

### DISCUSSION THREAD

**Discussion by lostsoul6** [XTLS/Xray-core#2054] [General]
Hello Friends ,

Unfortunately a couple of hours ago one of the major providers in Iran called IR-MCI blocked REALITY . At first I thought that maybe my VPS IP address is blocked but that's not the case and I can successfully SSH to the server . I was using yahoo.com as DEST but it's not working anymore . It seems GFW is updated and now REALITY is useless . Will there be any updates coming for REALITY to counter this?

Regards ,

**Comment by nursery01** [comment]
yahoo.com IPs white list？

Have you ever tried empty sni?

**Comment by lostsoul6** [comment]
Yahoo.com used to be a whitelisted SNI but they blocked it yesterday . Now the website opens but we can't use it as DEST for REALITY . it won't work with yahoo. 

What do you mean by empty SNI ? Isn't the whole point of REALITY faking TLS fingerprints of DEST so GFW can't block it by pattern? What will empty SNI do in combination with REALITY?

**Comment by nursery01** [comment]
> What do you mean by empty SNI ? Isn't the whole point of REALITY faking TLS fingerprints of DEST so GFW can't block it by pattern? What will empty SNI do in combination with REALITY?

Empty sni can invalidate the sni blacklist mechanism, this is just a method, and it does not mean that it is effective for a long time

The tls fingerprint is not calculated by sni

**Reply by ssmetall** [reply]
> > What do you mean by empty SNI ? Isn't the whole point of REALITY faking TLS fingerprints of DEST so GFW can't block it by pattern? What will empty SNI do in combination with REALITY?
> 
> Empty sni can invalidate the sni blacklist mechanism, this is just a method, and it does not mean that it is effective for a long time
> 
> The tls fingerprint is not calculated by sni

how to config this , i try empry sni but doesn't work

**Comment by ssmetall** [comment]
Hello Mr. rprx, help us, the speed of reality is very low, how to connect base on ip whitelist ? According to the investigations, the network does not whitelist based on SNI anymore

**Comment by lostsoul6** [comment]
@megasoheilsh this is my question as well . IR-MCI has blocked every SNI . No SNI can be used for reality now . We need another method . If anyone knows an SNI and DEST for IR-MCI please let me know .

**Reply by ssmetall** [reply]
> @megasoheilsh this is my question as well . IR-MCI has blocked every SNI . No SNI can be used for reality now . We need another method . If anyone knows an SNI and DEST for IR-MCI please let me know .

The problem is that they have turned off the sni whitelist and slowed down any type of IP

**Comment by lostsoul6** [comment]
@megasoheilsh surely they have some kind of whitelist because their apps also use some foreign services . I mean in the end they will eventually block all internet and we would be like North Korea . Just wanted to use internet properly while we have it :(

**Comment by nursery01** [comment]
I‘m curious whether Iran’s home network and company‘s business network are the same? I mean international speed and number of whitelists.

**Reply by lostsoul6** [reply]
I once said this on twitter but people laughed at me . I think each ISP in Iran has it's own GFW ! I think there is a room in each ISP and the people who work there have the power to do anything they want regarding blocking and even ISP CEO can't say anything about it ! This is why the blockage is completely different in all ISPs . 
Right now the datacenter servers have the better access to internet . after those are home ADSL users with less limitations . The worst are the two major companies of MTN Irancell and IR-MCI also known as Hamrahe-Avval who provide simcard internet and data .

**Comment by lilcheti** [comment]
Ok so i created this telegram group https://t.me/howisuconnected to discuss about how we are connected on each isp

**Comment by Arzenow** [comment]
سلام. SNI ایرانسل جدید دارید ؟ ایرانسل از کار افتاد حتی روی تانل

**Comment by us254** [comment]
I think Iran does not have a direct equivalent of China Telecom's CN2 GIA service

1. **CN2 GIA (China Telecom Next Generation Carrying Network Global Internet Access)**: This is a premium network service offered by China Telecom. The "CN2" refers to their Next Generation Carrier Network, which is a separate backbone from the public internet most consumers use. It's designed to provide higher performance in terms of stability, lower latency, and higher speeds. GIA specifically refers to the global internet access service over this network, catering to businesses that require reliable and fast connectivity to international internet resources.

2. **IPLC (International Private Leased Circuit)**: This is a point-to-point private line used by businesses to communicate between offices that are geographically dispersed across countries. It's essentially a reserved circuit and does not traverse the public internet, providing a secure and reliable connection with consistent bandwidth. This is particularly important for applications that are sensitive to latency and require high uptime.

3. **IEPL (International Ethernet Private Line)**: Similar to IPLC, IEPL provides a private point-to-point or point-to-multipoint service. However, it uses Ethernet technology over a carrier's backbone, which can often be more cost-effective and flexible in terms of bandwidth options compared to traditional IPLC. It is suitable for data-intensive applications such as business data exchange, video conferencing, and secure file transfers.

Both CN2 GIA and IPLC/IEPL services are designed to cater to businesses that need robust and dependable connectivity solutions, but they serve slightly different needs. CN2 GIA is more about high-quality access to the global internet, while IPLC/IEPL services are about connecting specific points with private, secure lines.
